src/AutoMapper/IProfileExpression.cs

```diff
@@ -53,6 +55,14 @@ public interface IProfileExpression
         IMappingExpression CreateMap(Type sourceType, Type destinationType, MemberList memberList);
 
         /// <summary>
+        /// Transforms PropertyMap with source and destination Type
+        /// </summary>
+        /// <typeparam name="TSource"></typeparam>
+        /// <typeparam name="TDestination"></typeparam>
+        /// <param name="transformExpression"></param>
+        void TransformPropertyMap<TSource>(Expression<Func<TSource, TSource>> transformExpression);

```

Comment:
I use `ReplaceParameters` so it has to be an expression.  You should be able to use `?.` or `??` or `? :` in the expressions.

I guess with conditions an example of what you want would be in order.  With more expression trickery I could get the PropertyMap but you already have code for that.